Makefile

```diff
@@ -73,6 +73,9 @@ lint: licensecheck go-lint
 # generates CRD using controller-gen
 crd: controller-gen
 	${CONTROLLER_GEN} crd:crdVersions=v1 paths="./endpoint/..." output:crd:stdout > docs/contributing/crd-source/crd-manifest.yaml
+	@echo -n "{{- if .Values.deployCrd }}" > charts/external-dns/templates/crd.yaml
+	cat docs/contributing/crd-source/crd-manifest.yaml >> charts/external-dns/templates/crd.yaml
+	@echo "{{- end }}"  >> charts/external-dns/templates/crd.yaml
```

Comment:
I'd rather this be a curated manual step, auto generated CRDs have noise which doesn't want to leak into the Helm chart.